src/Donations/Migrations/MoveDonationCommentToDonationMetaTable.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Give\Donations\Migrations;
+
+use Exception;
+use Give\Framework\Database\DB;
+use Give\Framework\Migrations\Contracts\Migration;
+use Give\Log\Log;
+
+/**
+ * Class MoveDonationCommentToDonationMetaTable
+ *
+ * @unreleased
+ */
+class MoveDonationCommentToDonationMetaTable extends Migration
+{
+    /**
+     * @inheritdoc
+     * @throws Exception
+     */
+    public function run()
+    {
+        $commentMetaTable = DB::prefix('give_commentmeta');
+        $commentTable = DB::prefix('give_comments');
+        $donationMetaTable = DB::prefix('give_donationmeta');
+
+        $commentsCount = DB::get_var(
+            "
+            SELECT
+                COUNT(1)
+            FROM
+                $commentTable
+            WHERE
+                comment_type = 'donor_donation'
+            "
+        );
+
+        if ( ! intval($commentsCount)) {
+            return;
+        }
+
+        DB::query('START TRANSACTION');
+
+        try {
+            $insertQuery = DB::query(
+                "
+                INSERT INTO $donationMetaTable (donation_id, meta_key, meta_value)
+                SELECT
+                    comment_parent,
+                    '_give_donation_comment',
+                    comment_content
+                FROM
+                    $commentTable
+                WHERE
+                    comment_type = 'donor_donation'
+                "
+            );
+
+            if ( ! $insertQuery) {
+                throw new Exception('Failed to insert donation comment into donation meta table.');
+            }
+
+            DB::query(
+                "
+                DELETE FROM $commentTable
+                WHERE comment_type = 'donor_donation'
+                    AND comment_parent IN(
+                        SELECT
+                            donation_id FROM $donationMetaTable
+                        WHERE
+                            meta_key = '_give_donation_comment')
+                "
+            );
+
+            DB::query(
+                "
+                DELETE FROM $commentMetaTable
+                WHERE give_comment_id IN(
+                        SELECT
+                            comment_ID FROM $commentTable
+                        WHERE
+                            comment_type = 'donor_donation'
+                            AND comment_parent IN(
+                                SELECT
+                                    donation_id FROM $donationMetaTable
+                                WHERE
+                                    meta_key = '_give_donation_comment'))
+                "
+            );
```

Comment:
I don't think you need to do the subquery `IN()` for the migrated comments. It seems like simply deleting based on the `comment_type` being `donor_donation` is sufficient, unless there are some comments of that type we intend to leave? But it doesn't seem so by the INSERT query.

Reply:
Resolved 124debc8b8457d7b14f7a378af4df282b37f2e86

I also had to reverse the order as the meta data needs the original comment to know what is its type.